### PR TITLE
fix: Format and log audit-level messages only when audit logging is enabled.

### PR DESCRIPTION
**Pull request by tippmar-nr**
Thank you for submitting a pull request.  Please review our [contributing guidelines](/CONTRIBUTING.md) and [code of conduct](https://opensource.newrelic.com/code-of-conduct/).

## Description
Adds code to check whether audit logging is enabled before going to the trouble of formatting audit log messages and sending them to Serilog. 

(Note that Serilog would already suppress sending those messages if audit logging was disabled, so there's no functional change with this PR, only an optimization to reduce unnecessary string formatting and Serilog calls.)

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

**Review by nr-ahemsath — APPROVED** [review]
Looks good! 👍

File: src/Agent/NewRelic/Agent/Core/Logging/AuditLog.cs [hunk 1]
```diff
@@ -13,21 +13,23 @@ public static class AuditLog
...
         {
+            if (IsAuditLogEnabled)
                 // use Fatal log level to ensure audit log messages never get filtered due to level restrictions
                 _lazyAuditLogger.Value.Fatal(message);
         }
 
         internal static LoggerConfiguration IncludeOnlyAuditLog(this LoggerConfiguration loggerConfiguration)
         {
-            return loggerConfiguration.Filter.ByIncludingOnly($"{LogLevelExtensions.AuditLevel} is not null");
+            IsAuditLogEnabled = true; // set a flag so we can short-circuit when audit log is not enabled
```
Comment (nrcventura) [inline src/Agent/NewRelic/Agent/Core/Logging/AuditLog.cs:30]: Should we add a test that ensures `IsAuditLogEnabled` is true when this method is called?

File: src/Agent/NewRelic/Agent/Core/DataTransport/HttpCollectorWire.cs [hunk 2]
```diff
@@ -156,10 +156,14 @@ public string SendData(string method, ConnectionInfo connectionInfo, string seri
...
-        private static void AuditLog(Direction direction, Source source, string uri)
+        private void AuditLog(Direction direction, Source source, string uri)
         {
-            var message = string.Format(AuditLogFormat, direction, source, Strings.ObfuscateLicenseKeyInAuditLog(uri, LicenseKeyParameterName));
-            Logging.AuditLog.Log(message);
+            if (Logging.AuditLog.IsAuditLogEnabled)
+            {
+                var message = string.Format(AuditLogFormat, direction, source,
+                    Strings.ObfuscateLicenseKeyInAuditLog(uri, LicenseKeyParameterName));
+                Logging.AuditLog.Log(message);
```
Comment (nrcventura) [inline src/Agent/NewRelic/Agent/Core/DataTransport/HttpCollectorWire.cs:165]: Should we add a test that verifies that we only attempt to generate the audit log when the AuditLogIsEnabled?

Comment (tippmar-nr) [reply src/Agent/NewRelic/Agent/Core/DataTransport/HttpCollectorWire.cs:165]: Done.

**Pull request merged**